### PR TITLE
🌱  Add link checker target for external links

### LINKED ISSUE #2703: liche, link checker should be more relaxed
Our link checker, https://github.com/raviqqe/liche, should not be checking links we don't control. This is asking to break CI whenever external links change.

However, we really shouldn't have any 404s so when external links change maybe it's good that it fails.

Unfortunately https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/docs/getting-started.md this moved. I think this should link to the repo itself where they would be responsible for making documentation obvious. It's less likely a repo will move than an actual document.

Separately, liche really needs to handle retries. https://github.com/raviqqe/liche/issues/36

/kind bug

### PR DESCRIPTION
**Pull request by benmoss**
**What this PR does / why we need it**:
We stopped checking external links in #2706, I believe because it was PR-blocking and would frequently fail if an external link started breaking, resulting it all PRs being blocked until that link was fixed.

This adds a separate target, `make verify-external-links`, which is not part of `make verify` and which we can add a separate non-blocking Prow job for.

I switched us to [mlc](https://github.com/becheran/mlc/) which supports throttling, which was another reason I think that the earlier external link checking was a problem. It's written in Rust so I'm using the Docker image rather than building it from source.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2703

### CODE REVIEW

**Comment by benmoss** [comment]
Ah dang we don't have docker in prow? hrm

**Comment by detiber** [comment]
The job will need to be modified to run as privileged and configured for `preset-dind-enabled`, similar to this job: https://github.com/kubernetes/test-infra/blob/d776361785430f0305f8ce489722f8e9f2b91d71/config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits.yaml#L20-L43

File: Makefile [hunk 1]
```diff
@@ -46,8 +46,7 @@ E2E_FRAMEWORK_DIR := test/framework
 CAPD_DIR := test/infrastructure/docker
 RELEASE_NOTES_BIN := bin/release-notes
 RELEASE_NOTES := $(TOOLS_DIR)/$(RELEASE_NOTES_BIN)
-LINK_CHECKER_BIN := bin/liche
-LINK_CHECKER := $(TOOLS_DIR)/$(LINK_CHECKER_BIN)
+LINK_CHECKER := docker run -t --rm -v "$$(pwd)\:/capi" becheran/mlc:0.13.4 mlc
```
Comment (ncdc) [inline Makefile:49]: I'd rather not run a third-party image if we can avoid it. WDYT @vincepri @detiber?

**Comment by benmoss** [comment]
There is a Linux binary released for this tool, but no OSX binary. We could release our own, but that's a hassle. We can also mirror the image.

File: Makefile (hunk 1 above)
Comment (benmoss) [reply Makefile:49]: What's the difference between a third party image and third-party go code?

Comment (ncdc) [reply Makefile:49]: This image doesn't have any details about what's inside it / how it was built (https://hub.docker.com/r/becheran/mlc). I assume it comes from https://github.com/becheran/mlc but there are no guarantees. Maybe I'm being overly cautious?

Comment (vincepri) [reply Makefile:49]: +1, I'd rather build the binary from a specific source then using a custom docker image

Comment (vincepri) [reply Makefile:49]: The linked repository seems pretty new, I'm not 100% comfortable to download a binary from a not-known tool to everyone's computer.

/hold

Comment (benmoss) [reply Makefile:49]: Lemme see what I can do

**Comment by vincepri** [comment]
From https://github.com/kubernetes/website/issues/4004, it was suggested to use https://github.com/raviqqe/muffet which publishes binaries for most platforms, and it's in Go, which means we can compile it.

**Comment by benmoss** [comment]
Cool, I hadn't come across that one in my searches

**Comment by benmoss** [comment]
Ah, muffet is for checking/crawling a running server, not markdown files

**Comment by vincepri** [comment]
We could make it a periodic prow job that checks against https://cluster-api.sigs.k8s.io

**Comment by benmoss** [comment]
Will reopen when I have a better plan

**Pull request closed without merging**